product/src/main/java/com/yas/product/repository/ProductRepository.java

```diff
@@ -40,22 +40,26 @@ Page<Product> getProductsWithFilter(@Param("productName") String productName,
     List<Product> findAllByIdIn(List<Long> productIds);
 
     @Query(value = "FROM Product p WHERE p.isFeatured = TRUE "
-            + "AND p.isVisibleIndividually = TRUE "
-            + "AND p.isPublished = TRUE ORDER BY p.id ASC ")
-    Page<Product> getFeaturedProduct(Pageable pageable);
+        + "AND p.isVisibleIndividually = TRUE "
+        + "AND p.isPublished = TRUE "
+        + "AND p.id IN (:productIds) "
```

Comment:
Please move `p.id IN (:productIds)` to the top condition to utilize the index on the `id` column.
Passing a large list of `productIds` in the IN clause may result in a query limit error.
Consider applying batching here.